File: services/content-service/src/main/java/com/novix/content_service/service/SearchService.java
package com.novix.content_service.service;

import com.novix.content_service.dto.request.SearchRequest;
import com.novix.content_service.dto.response.SearchResponse;

public interface SearchService {
    SearchResponse search(SearchRequest request);
    SearchResponse searchByCategory(String category, int page, int size);
}

File: frontend/web/src/services/search.service.ts
import api from '@/lib/api';
import { SearchResponse } from '@/types';

export const searchService = {
  async search(query: string, page = 0, size = 20): Promise<SearchResponse> {
    const response = await api.get<SearchResponse>(
      `/api/v1/search?query=${encodeURIComponent(query)}` + `&page=${page}&size=${size}`,
    );
    return response.data;
  },

  async searchByCategory(category: string, page = 0, size = 20): Promise<SearchResponse> {
    const response = await api.get<SearchResponse>(
      `/api/v1/search/category/${category}` + `?page=${page}&size=${size}`,
    );
    return response.data;
  },
};
